internal/provider/resource_gitlab_deploy_key_enable_test.go

```diff
@@ -2,145 +2,131 @@ package provider
 
 import (
 	"fmt"
-	"strconv"
-	"strings"
 	"testing"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
-	gitlab "github.com/xanzy/go-gitlab"
+	"github.com/xanzy/go-gitlab"
 )
 
 func TestAccGitlabDeployKeyEnable_basic(t *testing.T) {
-	var deployKey gitlab.ProjectDeployKey
-	rInt := acctest.RandInt()
+	testAccCheck(t)
+
+	testProjectParent := testAccCreateProject(t)
+	testProjectKeyShared := testAccCreateProject(t)
 
-	keyTitle := "main"
 	key := "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDblguSWgpqiXIjHPSas4+N3Dten7MTLJMlGQXxGpaqN9nGPdNmuRB2YXyjT/nrryoY/qrtuVkPnis5WVo8N/s3hAnJbeJPUS2WKEGjpBlL34AQ+ANnlmGY8L6zr82Hp2Ommb7XGGtlq5D3yLCgTfcXLjC51tgcdwHsdH1U+RisgLwaTSrP/HF4G7IAr5ATsyYjtCwQRQ8ijdf5A34+XN6h8J6TLXKab5eZDuH38s9LxJuS7MRxx/P2UTOsqfjtrZWoQgE5adEGvnDxKyruex9PzNbCNVahzsma7tdikDbzxlHLIZ1aht6rKuai3iyLgcZfGIYtkq4xvg/bnNXxSsGf worker@kg.getwifi.com"
 
+	canPushDeployKeyOptions := gitlab.AddDeployKeyOptions{
+		Title:   gitlab.String("main"),
+		Key:     gitlab.String(key),
+		CanPush: gitlab.Bool(true),
+	}
+
+	parentProjectDeployKey := testAccCreateDeployKey(t, testProjectParent.ID, &canPushDeployKeyOptions)
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
 		ProviderFactories: providerFactories,
 		CheckDestroy:      testAccCheckGitlabDeployKeyEnableDestroy,
 		Steps: []resource.TestStep{
-			// Create a project and deployKey with default options
+			// Enable a deployKey on project with default options
+			{
+				Config: testAccGitlabDeployKeyEnableConfig(testProjectKeyShared.ID, parentProjectDeployKey.ID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("gitlab_deploy_key_enable.foo", "project"),
```

Comment:
since this is not a computed field and there is a subsequent `Verify import` test step it's superfluous ...

Reply:
... same for the other test steps ..

Reply:
Would you recommend we just remove the one check for project then across the test steps then is all?

Reply:
Yes, but for all non-computed fields - thus, `can_push` is the same or the `title` in the `gitlab_deploy_key` resource ...

Reply:
Check if the commit I just pushed is what you were thinking when you have time

Reply:
... yes, right direction - however, the `can_push` and `title` etc. is still there, which can also be removed :)

Reply:
Since there are no computed fields for this resource should I do just configs with no attribute checks and then just have import verifies in-between the test steps?

Reply:
I have removed the non-computed fields from attr checks in the acceptance tests and added the title and key to the enable resource tests